CMake/mlpack.cmake

```
@@ -592,6 +592,54 @@ macro(find_mlpack_internal)
 
 endmacro()
 
+macro(compile_OpenBLAS)
+  if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
+    set(OPENBLAS_SRC_DIR ${CMAKE_BINARY_DIR}/deps/OpenBLAS-${OPENBLAS_VERSION})
+    set(OPENBLAS_BUILD_DIR ${OPENBLAS_SRC_DIR}/build)
+    set(OPENBLAS_OUTPUT_LIB_DIR ${OPENBLAS_BUILD_DIR}/lib)
+    # always compile BLAS as release.
+    set(BLASS_BUILD_TYPE "Release")
+
+    if (CMAKE_GENERATOR MATCHES "Visual Studio")
+      set(OPENBLAS_OUTPUT_LIB_DIR ${OPENBLAS_BUILD_DIR}/lib/RELEASE)
+    endif()
+    if (CMAKE_GENERATOR MATCHES "Ninja")
+      set(OPENBLAS_OUTPUT_LIB_DIR ${OPENBLAS_BUILD_DIR}/lib/Release)
+    endif()
+
+    if (NOT EXISTS "${OPENBLAS_OUTPUT_LIB_DIR}/openblas.lib")
+      message(STATUS "Compiling OpenBLAS")
+      file(MAKE_DIRECTORY ${OPENBLAS_BUILD_DIR})
+      # -G -A -T to pass settings from current cmake command.
+      execute_process(
+              COMMAND ${CMAKE_COMMAND}
+              -G "${CMAKE_GENERATOR}"
+              -A "${CMAKE_GENERATOR_PLATFORM}"
+              -T "${CMAKE_GENERATOR_TOOLSET}"
+              "-DCMAKE_BUILD_TYPE=${BLASS_BUILD_TYPE}"
```

```suggestion
              "-DCMAKE_BUILD_TYPE=${BLAS_BUILD_TYPE}"
```
(To go with the suggestion above.)